caas/kubernetes/provider/k8s.go

```diff
@@ -180,6 +180,12 @@ func (k *kubernetesClient) SetConfig(cfg *config.Config) error {
 	return nil
 }
 
+// SetCloudSpec is specified in the environs.Environ interface.
+func (k *kubernetesClient) SetCloudSpec(spec environs.CloudSpec) error {
```

Comment:
given that this seems optional, why implement a no-op version?
if we *didn't* implement it, couldn't we just then check if it was supported and if not, do the no-op in the caller?
That's what we do for the NetworkingEnviron code

Reply:
yeah. i should have done that and will fix